### PR TITLE
Add Creality Board 452 Support

### PR DESCRIPTION
**Pull request by InsanityAutomation**
Following work by @Sebazzz on the Creality CR6 printer, this breaks out the motherboard support to give a path to support in chunks, minimizing rebase work and potential conflicts as LCD code evolves.

Probes with explicit enable pin coming up next followed by ExtUI is homing extensions.

### CODE REVIEW

File: Marlin/src/HAL/STM32F1/HAL.h [hunk 1]
```diff
@@ -109,6 +109,8 @@
   #else
     #error "LCD_SERIAL_PORT must be -1 or from 1 to 3. Please update your configuration."
   #endif
+
+  #define SERIAL_GET_TX_BUFFER_FREE LCD_SERIAL.availableForWrite
```
Comment (sjasonsmith) [inline Marlin/src/HAL/STM32F1/HAL.h:113]: Is this change necessary in this commmit? It seems unrelated to me.

File: Marlin/src/pins/stm32f1/pins_CREALITY_V452.h [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  #define IIC_BL24CXX_EEPROM                      // EEPROM on I2C-0 used only for display settings
+  #if ENABLED(IIC_BL24CXX_EEPROM)
+    #define IIC_EEPROM_SDA                  PA11
+    #define IIC_EEPROM_SCL                  PA12
+    #define MARLIN_EEPROM_SIZE             0x800  // 2Kb (24C16)
+  #else
+    #define SDCARD_EEPROM_EMULATION               // SD EEPROM until all EEPROM is BL24CXX
+    #define MARLIN_EEPROM_SIZE             0x800  // 2Kb
+  #endif
+#endif
```
Comment (sjasonsmith) [inline Marlin/src/pins/stm32f1/pins_CREALITY_V452.h]: I just reworked these in pins_Creaity_V4, so that people could choose to use SD EEPROM if they want. This is what I ended up with:

```
//
// EEPROM
//
#if NO_EEPROM_SELECTED
  #define IIC_BL24CXX_EEPROM                      // EEPROM on I2C-0
  //#define SDCARD_EEPROM_EMULATION
#endif

#if ENABLED(IIC_BL24CXX_EEPROM)
  #define IIC_EEPROM_SDA                  PA11
  #define IIC_EEPROM_SCL                  PA12
  #define MARLIN_EEPROM_SIZE             0x800  // 2Kb (24C16)
#elif ENABLED(SDCARD_EEPROM_EMULATION)
  #define MARLIN_EEPROM_SIZE             0x800  // 2Kb
#endif
```

File: Marlin/src/pins/stm32f1/pins_CREALITY_V452.h [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+
+//
+// Limit Switches
+//
+#define X_MIN_PIN           PC4
+// #define X_MAX_PIN          PA7
```
Comment (sjasonsmith) [inline Marlin/src/pins/stm32f1/pins_CREALITY_V452.h:58]: Does this board potentially have separate Min/Max endstops?

It seems like these should probably just define STOP pins.

File: Marlin/src/pins/stm32f1/pins_CREALITY_V452.h [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+#define Y_MIN_PIN           PC5
+#define Z_MIN_PIN           PA4
+#define COM_PIN             PA5
```
Comment (sjasonsmith) [inline Marlin/src/pins/stm32f1/pins_CREALITY_V452.h]: What is this? Does this have something to do with the probe?

File: Marlin/src/pins/stm32f1/pins_CREALITY_V452.h [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+#define SDIO_SUPPORT      // Extra added by Creality
+#define SDIO_CLOCK 6000000 // In original source code overridden by Creality in sdio.h
+
+#define CASE_LIGHT_PIN PA6
+
+#define FIL_RUNOUT_PIN PA7
+#define PROBE_ENABLE_PIN    PC6 // Optoswitch to Enable Z Probe
+
+
+#define TEMP_TIMER_CHAN 4 // Channel of the timer to use for compare and interrupts
```
Comment (sjasonsmith) [inline Marlin/src/pins/stm32f1/pins_CREALITY_V452.h]: I doubt this is actually needed.

File: Marlin/src/pins/stm32f1/pins_CREALITY_V452.h (hunk 4 above)
Comment (sjasonsmith) [reply Marlin/src/pins/stm32f1/pins_CREALITY_V452.h]: This specifically refers to `COM_PIN`. I'm just not sure what it is.

File: Marlin/src/pins/stm32f1/pins_CREALITY_V452.h (hunk 5 above)
Comment (sjasonsmith) [reply Marlin/src/pins/stm32f1/pins_CREALITY_V452.h]: Referring to `TEMP_TIMER_CHAN`

File: Marlin/src/HAL/STM32F1/HAL.h (hunk 1 above)
Comment (InsanityAutomation) [reply Marlin/src/HAL/STM32F1/HAL.h:113]: This could go in either commit really but will be beneficial for any serial display on this board so i dropped it here.

File: Marlin/src/pins/stm32f1/pins_CREALITY_V452.h (hunk 4 above)
Comment (InsanityAutomation) [reply Marlin/src/pins/stm32f1/pins_CREALITY_V452.h]: Yes, ill be renaming it to a TARE pin as I generalize that. When the next PR is finished ill come back and modify this.

File: Marlin/src/pins/stm32f1/pins_CREALITY_V452.h (hunk 5 above)
Comment (InsanityAutomation) [reply Marlin/src/pins/stm32f1/pins_CREALITY_V452.h]: Maybe, I would need to follow it back a bit but for the moment all known references have it so I didnt touch it till I had time to investigate.

File: Marlin/src/pins/stm32f1/pins_CREALITY_V452.h (hunk 2 above)
Comment (InsanityAutomation) [reply Marlin/src/pins/stm32f1/pins_CREALITY_V452.h]: Ill take a closer look if this lines up here as well when the sensor stuff is done.

File: Marlin/src/pins/stm32f1/pins_CREALITY_V452.h (hunk 3 above)
Comment (InsanityAutomation) [reply Marlin/src/pins/stm32f1/pins_CREALITY_V452.h:58]: I havnt pulled the cover off yet so ill let you know :)

File: Marlin/src/pins/stm32f1/pins_CREALITY_V452.h (hunk 2 above)
Comment (sjasonsmith) [reply Marlin/src/pins/stm32f1/pins_CREALITY_V452.h]: I went ahead and changed this. There is no functional change, other than allowing SDCARD_EEPROM_EMULATION from a config file to properly set MARLIN_EEPROM_SIZE.

File: Marlin/src/pins/stm32f1/pins_CREALITY_V452.h (hunk 4 above)
Comment (sjasonsmith) [reply Marlin/src/pins/stm32f1/pins_CREALITY_V452.h]: This is in my push, based on our chat.

File: Marlin/src/pins/stm32f1/pins_CREALITY_V452.h (hunk 5 above)
Comment (sjasonsmith) [reply Marlin/src/pins/stm32f1/pins_CREALITY_V452.h]: It looks like it is already in the PIO environment, so I removed it here.
It probably isn't _actually_ needed anywhere, but removing it here should not matter.
